frontend/src/views/Settings/OrgSettingsPage/components/AppConnectionsTab/components/AppConnectionForm/GitHubConnectionForm.tsx

```diff
@@ -0,0 +1,172 @@
+import crypto from "crypto";
+
+import { useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+
+import { Button, FormControl, Input, ModalClose, Select, SelectItem } from "@app/components/v2";
+import { APP_CONNECTION_MAP, APP_CONNECTION_METHOD_MAP } from "@app/helpers/appConnections";
+import {
+  GitHubConnectionMethod,
+  TGitHubConnection,
+  useGetAppConnectionOption
+} from "@app/hooks/api/appConnections";
+import { AppConnection } from "@app/hooks/api/appConnections/enums";
+import { slugSchema } from "@app/lib/schemas";
+
+type Props = {
+  appConnection?: TGitHubConnection;
+};
+
+const rootSchema = z.object({
+  name: slugSchema({ min: 1, max: 32, field: "Name" }),
+  app: z.literal(AppConnection.GitHub)
+});
+
+const formSchema = z.discriminatedUnion("method", [
+  rootSchema.extend({
+    method: z.literal(GitHubConnectionMethod.App)
+  }),
+  rootSchema.extend({
+    method: z.literal(GitHubConnectionMethod.OAuth)
+  })
+]);
+
```

Comment:
is discriminated union still necessary here?

Reply:
no 🤡